spec/unit/pdk/validate/puppet/puppet_epp_spec.rb

```diff
@@ -0,0 +1,319 @@
+      expect(described_class.pattern_ignore).to eq('')
+    end
+  end
+
+  describe '.invoke' do
+    context 'when the validator runs correctly' do
+      before(:each) do
+        allow(described_class).to receive(:parse_targets).with(anything).and_return([[], [], []])
+      end
+
+      it 'cleans up the temp dir after invoking' do
+        expect(described_class).to receive(:remove_validate_tmpdir)
+        described_class.invoke(PDK::Report.new, {})
+      end
+    end
+
+    context 'when the validator raises an exception' do
+      before(:each) do
+        allow(described_class).to receive(:parse_targets).with(anything).and_raise(PDK::CLI::FatalError)
+      end
+
+      it 'cleans up the temp dir after invoking' do
+        expect(described_class).to receive(:remove_validate_tmpdir)
+        expect {
+          described_class.invoke(PDK::Report.new, {})
+        }.to raise_error(PDK::CLI::FatalError)
+      end
+    end
+  end
+
+  describe '.remove_validate_tmpdir' do
+    after(:each) do
+      described_class.remove_validate_tmpdir
+    end
+
+    context 'when no temp dir has been created' do
+      before(:each) do
+        described_class.instance_variable_set('@validate_tmpdir', nil)
+      end
+
+      it 'does not attempt to remove the directory' do
+        expect(FileUtils).not_to receive(:remove_entry_secure)
+      end
+    end
+
+    context 'when a temp dir has been created' do
+      before(:each) do
+        described_class.validate_tmpdir
+        allow(File).to receive(:directory?).and_call_original
+      end
+
+      context 'and the path is a directory' do
+        before(:each) do
+          allow(File).to receive(:directory?).with(tmpdir).and_return(true)
+        end
+
+        it 'removes the directory' do
+          expect(FileUtils).to receive(:remove_entry_secure).with(tmpdir)
+        end
+      end
+
+      context 'but the path is not a directory' do
+        before(:each) do
+          allow(File).to receive(:directory?).with(tmpdir).and_return(false)
+        end
+
+        it 'does not attempt to remove the directory' do
+          expect(FileUtils).not_to receive(:remove_entry_secure)
+        end
+      end
+    end
+  end
+
+  describe '.parse_options' do
+    subject(:command_args) { described_class.parse_options(options, targets) }
+
+    let(:options) { {} }
+    let(:targets) { %w[target1 target2.epp] }
+
+    before(:each) do
+      allow(Gem).to receive(:win_platform?).and_return(false)
+    end
+
+    it 'invokes `puppet epp validate`' do
+      expect(command_args.first(2)).to eq(%w[epp validate])
+    end
+
+    it 'appends the targets to the command arguments' do
+      expect(command_args.last(targets.count)).to eq(targets)
+    end
+
+    context 'when auto-correct is enabled' do
+      let(:options) { { auto_correct: true } }
+
+      it 'has no effect' do
+        expect(command_args).to eq(%w[epp validate --config /dev/null --modulepath].push(tmpdir).concat(targets))
```

Comment:
Note for me, not to stop merging:  Need to investigate why using a fixed Linux path.  This shouldn't work on Windows?